internal/sirius/client.go

```diff
@@ -39,7 +40,12 @@ func (c *Client) newRequest(ctx context.Context, method, path string, body io.Re
 		}
 	}
 	req.Header.Add("OPG-Bypass-Membrane", "1")
-	req.Header.Add("X-XSRF-TOKEN", xsrfToken)
+	headerToken, _ := url.QueryUnescape(xsrfToken)
+	req.Header.Add("X-XSRF-TOKEN", headerToken)
+
+	if method == http.MethodPut || method == http.MethodPost {
```

Comment:
Not necessarily the case as the change password request POSTs a form, this method returns a `*http.Request` so you can add headers after calling it instead of overloading `newRequest` with ifs

Reply:
👍